### PR TITLE
Iterators

### PR DESCRIPTION
**Pull request by sdcooke**

### CODE REVIEW

File: tinyquery/evaluator.py [hunk 1]
```diff
@@ -279,7 +279,7 @@ def evaluate_within(self, select_fields, group_set, ctx,
             ctx_with_primary_key = context.empty_context_from_template(ctx)
             context.append_context_to_context(ctx, ctx_with_primary_key)
 
-            (table_name, _), _ = ctx_with_primary_key.columns.items()[0]
```
Comment (csilvers) [inline tinyquery/evaluator.py:282]: Wha??  Thanks for cleaning this up.

File: tinyquery/lexer.py [hunk 2]
```diff
@@ -63,7 +63,7 @@
     'FLOAT',
     'ID',
     'STRING'
-] + reserved_words.values()
+] + list(reserved_words.values())  # wrapping with list() to support python 3
```
Comment (csilvers) [inline tinyquery/lexer.py:66]: Thanks for that comment!

File: tinyquery/tinyquery.py [hunk 3]
```diff
@@ -143,7 +143,7 @@ def flatten_row(output, row, schema, prefix='', ever_repeated=False):
             return output
 
         def process_row(row):
-            for (key, value) in row.iteritems():
+            for (key, value) in row.items():
```
Comment (csilvers) [inline tinyquery/tinyquery.py:146]: This is the one place we iterate over a row, right?  So this could be big?

**Review by csilvers — APPROVED** [review]
I'm more or less ok with this one -- I agree that it seems like not-iterating in python2 shouldn't be such a big deal for these.

One thing I'd like to see once this is squashed is a better commit message.  In particular, it should talk about why we're moving from iterfoo to foo (to move closer to support python3) and why it's ok performance-wise (almost all uses are on columns or other iterables that we expect to be small).

My biggest worry is the one place where -- I think -- we are iterating over rows.  But maybe even in that case we're iterating over all the columns in a row?  If you think that use is ok, it's good by me.

File: tinyquery/tinyquery.py (hunk 3 above)
Comment (sdcooke) [reply tinyquery/tinyquery.py:146]: It's iterating over the columns in a row, so should be similar in size to any other column iterators

**Pull request closed without merging**